test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/PopFrameTest.java

```diff
@@ -0,0 +1,252 @@
+            testTask.sleep(5);
+        }
+
+        log("\nMain #B: method B() must be blocked in a breakpoint event handler");
+        {
+            ensureAtBreakpoint();
+
+            log("\nMain #B.1: unsuspended");
+            errCode = popFrame(testTaskThread);
+            if (errCode != THREAD_NOT_SUSPENDED) {
+                throwFailed("Main #B.1: expected THREAD_NOT_SUSPENDED instead of: " + errCode);
+            }
+            log("Main #B.1: got expected THREAD_NOT_SUSPENDED");
+
+            log("\nMain #B.2: suspended");
+            suspendThread(testTaskThread);
+            errCode = popFrame(testTaskThread);
+            if (errCode != JVMTI_ERROR_NONE) {
+                throwFailed("Main #B.2: expected JVMTI_ERROR_NONE");
+            }
+            log("Main #B.2: got expected JVMTI_ERROR_NONE");
+            resumeThread(testTaskThread);
+            notifyAtBreakpoint();
+
+            log("\nMain #B.3: unsuspended, call PopFrame on own thread");
+            ensureAtBreakpoint();
+            notifyAtBreakpoint();
+            testTask.sleep(5);
+        }
+
+        log("\nMain #C: method C() calls PopFrame on its own thread");
+        {
+            // PopFrame is called from the test task (own thread) and expected to succeed.
+            // No suspension of the test task thread is required or can be done in this case.
+            testTask.ensureFinished();
+        }
+
+        try {
+            testTaskThread.join();
+        } catch (InterruptedException ex) {
+            throwFailed("Unexpected " + ex);
+        }
+    }
+
+
+    static class TestTask implements Runnable {
+        static private volatile boolean doLoop = true;
+        static void log(String str) { System.out.println(str); }
+
+        private volatile boolean started = false;
+        private volatile boolean finished = false;
+
+        static public void sleep(long millis) {
+            try {
+                Thread.sleep(millis);
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Interruption in TestTask.sleep: \n\t" + e);
+            }
+        }
+
+        // Ensure thread is ready.
+        public void ensureStarted() {
+            while (!started) {
+                sleep(1);
+            }
+        }
+
+        // Ensure thread is finished.
+        public void ensureFinished() {
+            while (!finished) {
+                sleep(1);
+            }
+        }
+
+        static public void clearDoLoop() {
+            doLoop = false;
+        }
+
+        public void run() {
+            log("TestTask.run: started");
+            started = true;
+
+            A();
+            sleep(1); // to cause yield
+
+            B();
+            sleep(1); // to cause yield
+
+            B();
+            sleep(1); // to cause yield
+
+            C();
+            finished = true;
+        }
+
+        // Method is blocked on entering a synchronized statement.
```

Comment:
Not sure I see where this method is blocked.

Reply:
Good catch, thanks. The comment is a left over from the initial test version.